### PR TITLE
Make high-profile available 

### PR DESCRIPTION
**Pull request by shts**

### CODE REVIEW

**Comment by ffMathy** [comment]
@ypresto this fixes #72 which also leads to various issues with Cordova's video editor plugin, which is using your plugin and has this error (for instance, on OnePlus devices).

**Review by ffMathy — APPROVED** [review]

**Comment by maxpaj** [comment]
Can this be merged?

**Comment by natario1** [comment]
After trying with #71 some months ago I decided to go forward and make a fork which I am maintaining and has dozens of commits ahead. I have removed this profile limitation a long time ago.

If you want, come take a look at https://github.com/natario1/Transcoder . Docs are still missing, but the demo app will show the new API surface.

**Comment by ypresto** [comment]
Sorry for late response.
I have very very little time for coding outside of work because my :baby: is :running: :blush: :boom: :boom: :couple: :scream: :sob: :tired_face:

As [supported video format list](https://developer.android.com/guide/topics/media/media-formats), high profile encoder is not supported yet. This means there is no guarantee that codecs of all the devices will output valid video data.

This library highly depends on hardware codec impl, and sometimes nasty bug was produced like this:
https://github.com/ypresto/android-transcoder/issues/8 and some Nexus-4-era glitch issues.
For other hand high profile w/ 1080p might produce glitch even if 720p does not.
Important thing is that not all the devices use Snapdragon.

So I am conservative for profile and resolution support. This is also because this lib provides transcoding for the family album (yes, it is very precious memory which should never get glitch) app [mitene](https://mitene.us), which has over a million Android users.

But maybe you can check compatibility [like this](https://developer.android.com/reference/android/media/MediaCodecInfo.CodecProfileLevel, or by xml file stored in /etc (?) of device. Or by testing running codec before using it.

**Comment by coreysherman** [comment]
@ypresto I'm not quire sure I understand the response with regards to merging this branch.  As mentioned above, your plugin is used by Cordova's video editor plugin, which is also used by ionic native.  Are there any plan to merge this fix in?

**Comment by ypresto** [comment]
@coreysherman 

TL;DR:
I'm opposite to output high profile video without checking codec support, as codec can produce glitch.
Instead, we can transcode the video even if resolution is already small enough (e.g. 720p).

**Comment by ypresto** [comment]
I was misunderstanding that cordova video plugin [returns null from format strategy (so original AVC video data is used regardless of its profile)](https://github.com/ypresto/android-transcoder/blob/012c3ab81053c17eea0ec42f2ec815cb7ad311be/lib/src/main/java/net/ypresto/androidtranscoder/format/Android720pFormatStrategy.java#L68), but actually it does not.

So, in OnePlus 6 (I also have it) era device, the codec generates high-profile video even without passing any profile parameter. Is it right?

Profile validator was introduced because lollipop-era device does not well support non-baseline video.
See this nasty XXX line in libstagefright of lollipop, it overwrites profile value without throwing error :joy: ↓
https://android.googlesource.com/platform/frameworks/av/+/lollipop-release/media/libstagefright/OMXCodec.cpp#1226

My recommendation at lollipop-era was stick with baseline, but now we can just remove the validator, or would be better to check the profile is supported by CodecProfileLevel or etc.

File: lib/src/main/java/net/ypresto/androidtranscoder/engine/MediaFormatValidator.java [hunk 1]
```diff
@@ -36,7 +37,8 @@ public static void validateVideoOutputFormat(MediaFormat format) {
         }
         ByteBuffer spsBuffer = AvcCsdUtils.getSpsBuffer(format);
```
Comment (ypresto) [inline lib/src/main/java/net/ypresto/androidtranscoder/engine/MediaFormatValidator.java]: Please remove lines below here, as profile validation is already not necessary.
https://github.com/ypresto/android-transcoder/pull/73#issuecomment-480259378

Comment (shts) [reply lib/src/main/java/net/ypresto/androidtranscoder/engine/MediaFormatValidator.java]: @ypresto done! df7ee27

**Comment by pengkobe** [comment]
Any update?  i am waiting so long time...

**Comment by martinbertinat** [comment]
@pengkobe I'm waiting too.. But.. Can we help in any wey @ypresto ?

**Comment by todoscontrajuan** [comment]
It looks like this can be merge, @ypresto can you approve this?

**Comment by Maxgamerboy1** [comment]
Bump! I would also like to consume this change. Thanks!

**Comment by davidTaveras48** [comment]
@ypresto im developing on ionic and im having this error :
Non-baseline AVC video profile is not supported by Android OS, actual profile_idc: 100

just on android 9.x or higher.

any solution for this ?

thanks
